### PR TITLE
add authentication mechanism option

### PR DESCRIPTION
**Pull request by Rhymmor**
Just added the option to MongoConnectionOptions and the corresponding parameter to MongoClient's connect (from docs: http://mongodb.github.io/node-mongodb-native/3.0/api/MongoClient.html#.connect)

### CODE REVIEW

**Comment by pleerock** [comment]
Thank you very much!

File: src/driver/mongodb/MongoConnectionOptions.ts [hunk 1]
```diff
@@ -245,4 +245,8 @@ export interface MongoConnectionOptions extends BaseConnectionOptions {
      */
     readonly loggerLevel?: "error"|"warn"|"info"|"debug";
 
+    /**
+     * Specifies the list of authentication mechanisms the server accepts.
```
Comment (vlapo) [inline src/driver/mongodb/MongoConnectionOptions.ts]: Description is little bit strange. What about "**Sets the authentication mechanism that MongoDB will use to authenticate the connection.**". Also there is `docs/connection-options.md` file with all options.

Comment (Rhymmor) [reply src/driver/mongodb/MongoConnectionOptions.ts]: Oh, I forgot to change it after copy/paste. Sure, I fix it!

Comment (vlapo) [reply src/driver/mongodb/MongoConnectionOptions.ts]: Nice :)

**Pull request merged**